armi/reactor/reactors.py

```diff
@@ -2321,3 +2327,30 @@ def processLoading(self, cs, dbLoad: bool = False):
         self.p.maxAssemNum = self.getMaxParam("assemNum")
 
         getPluginManagerOrFail().hook.onProcessCoreLoading(core=self, cs=cs)
+
+    def _updateBlockBOLHeights(self, assems: list, dbLoad: bool, detAxExp: bool):
+        """expand assemblies, resolve disjoint axial mesh (if needed), and update block BOL heights"""
+        axialExpChngr = AxialExpansionChanger(detAxExp)
+        for a in assems:
+            if not a.hasFlags(Flags.CONTROL):
+                axialExpChngr.setAssembly(a)
+                axialExpChngr.expansionData.computeThermalExpansionFactors()
+                axialExpChngr.axiallyExpandAssembly(thermal=True)
+        # resolve axially disjoint mesh (if needed)
+        if not dbLoad:
+            axialExpChngr.manageCoreMesh(self.parent)
+            self.p.referenceBlockAxialMesh = self.findAllAxialMeshPoints(
+                applySubMesh=False
+            )
+            self.p.axialMesh = self.findAllAxialMeshPoints()
+        else:
```

Comment:
can be reduced to `elif`

Reply:
Addressed https://github.com/terrapower/armi/pull/823/commits/cc59a8f0e474953d2c57a6f588e2208027c3d5f4